### PR TITLE
charon release v0.13.0

### PR DESCRIPTION
**Pull request by xenowits**
- Updates charon cli references
- Creates a versioned docs folder for v0.13.0

### CODE REVIEW

File: docs/charon/charon_cli_reference.md [hunk 1]
```diff
@@ -183,36 +194,81 @@ Usage:
...
+      --p2p-max-reservations int       Updates max circuit reservations per peer (each valid for 30min) (default 512)
+      --p2p-relay-loglevel string      Libp2p circuit relay log level. E.g., debug, info, warn, error.
+      --p2p-relays strings             Comma-separated list of libp2p relay URLs or ENRs. (default [http://bootnode.lb.gcp.obol.tech:3640/enr])
+      --p2p-tcp-address strings        Comma-separated list of listening TCP addresses (ip and port) for libP2P traffic. Empty default doesn't bind to local port therefore only supports outgoing connections.
+      --p2p-udp-address string         Listening UDP address (ip and port) for discv5 discovery. Empty default disables discv5 discovery. Discv5 is deprecated, use relay discovery.
+
 ```
 
-## Host a bootnode
+## Host a bootnode (Deprecated)
```
Comment (xenowits) [inline docs/charon/charon_cli_reference.md:271]: Deprecated section. Needs to be deleted in future releases when charon ends support for `charon bootnode`

File: docs/charon/charon_cli_reference.md [hunk 2]
```diff
@@ -183,36 +194,81 @@ Usage:
...
+      --private-key-file string            The path to the charon enr private key file. (default ".charon/charon-enr-private-key")
+      --simnet-beacon-mock                 Enables an internal mock beacon node for running a simnet.
+      --simnet-validator-keys-dir string   The directory containing the simnet validator key shares. (default ".charon/validator_keys")
+      --simnet-validator-mock              Enables an internal mock validator client when running a simnet. Requires simnet-beacon-mock.
+      --synthetic-block-proposals          Enables additional synthetic block proposal duties. Used for testing of rare duties.
+      --validator-api-address string       Listening address (ip and port) for validator-facing traffic proxying the beacon-node API. (default "127.0.0.1:3600")
+
+```
+
+## Host a relay
```
Comment (xenowits) [inline docs/charon/charon_cli_reference.md:233]: New section added for relays

File: versioned_docs/version-v0.13.0/charon/charon_cli_reference.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      --p2p-udp-address string         Listening UDP address (ip and port) for discv5 discovery. Empty default disables discv5 discovery. Discv5 is deprecated, use relay discovery.
+
+```
+
+### Create a full cluster locally
+
+`charon create cluster` creates a set of distributed validators locally, including the private keys, a `cluster-lock.json` file, and deposit and exit data. However, this command should only be used for solo use of distributed validators. To run a Distributed Validator with a group of operators, it is preferable to create these artifacts using the `charon dkg` command. That way, no single operator custodies all of the private keys to a distributed validator.
+
+```markdown
+charon create cluster --help
```
Comment (xenowits) [inline versioned_docs/version-v0.13.0/charon/charon_cli_reference.md:97]: This is not yet merged into main yet - PR https://github.com/ObolNetwork/charon/pull/1622

File: docs/charon/charon_cli_reference.md (hunk 1 above)
Comment (OisinKyne) [reply docs/charon/charon_cli_reference.md:271]: Can we make that a lowercase D

**Comment by OisinKyne** [comment]
Hey @xenowits, we should probably split this PR into two steps, one for adding the new CLI reference and a second for running the `yarn run version` command. Because for example the quickstart guides have not been updated here, so any updates made to them would be too late for v0.13.0

**Comment by xenowits** [comment]
> Hey @xenowits, we should probably split this PR into two steps, one for adding the new CLI reference and a second for running the `yarn run version` command. Because for example the quickstart guides have not been updated here, so any updates made to them would be too late for v0.13.0
> 
> Also there are broken URLs in the CI/CD

Cool then, Im opne a PR containing the charon cli changes. Once merged, we can then do the final `yarn run version` PR

File: versioned_sidebars/version-v0.13.0-sidebars.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (OisinKyne) [inline versioned_sidebars/version-v0.13.0-sidebars.json:1]: Delete this file too

File: docs/charon/charon_cli_reference.md [hunk 5]
```diff
@@ -100,16 +103,20 @@ Usage:
 Flags:
       --clean                          Delete the cluster directory before generating it.
       --cluster-dir string             The target folder to create the cluster in. (default ".charon/cluster")
+      --definition-file string         Optional path to a cluster definition file or an HTTP URL. This overrides all other configuration flags.
       --fee-recipient-address string   Optional Ethereum address of the fee recipient
   -h, --help                           Help for cluster
-      --name string                    Optional cosmetic cluster name
+      --insecure-keys                  Generates insecure keystore files. This should never be used. It is not supported on mainnet.
+      --keymanager-addresses strings   Comma separated list of keymanager URLs to push validator key shares to. Note that multiple addresses are required, one for each node in the cluster, with node0's keyshares being pushed to the first address, node1's keyshares to the second, and so on.
```
Comment (OisinKyne) [inline docs/charon/charon_cli_reference.md:110]: Did we go through with the implementation approach or did we just go with a single keymanager for local? What happens if a random one of these requests fails?

File: docs/charon/charon_cli_reference.md [hunk 6]
```diff
@@ -100,16 +103,20 @@ Usage:
...
+      --name string                    The cluster name
       --network string                 Ethereum network to create validators for. Options: mainnet, gnosis, goerli, kiln, ropsten, sepolia. (default "goerli")
-  -n, --nodes int                      The number of charon nodes in the cluster. Minimum is 4. (default 4)
+      --nodes int                      The number of charon nodes in the cluster. Minimum is 4. (default 4)
```
Comment (OisinKyne) [inline docs/charon/charon_cli_reference.md:113]: Might have to be cautious we didn't break any old guides here if the -n flag is removed. Shouldn't be any old make files around any more, but maybe some sh scripts might have used the -n flag.

File: docs/charon/charon_cli_reference.md [hunk 7]
```diff
@@ -183,36 +194,81 @@ Usage:
...
+      --simnet-validator-keys-dir string   The directory containing the simnet validator key shares. (default ".charon/validator_keys")
+      --simnet-validator-mock              Enables an internal mock validator client when running a simnet. Requires simnet-beacon-mock.
+      --synthetic-block-proposals          Enables additional synthetic block proposal duties. Used for testing of rare duties.
+      --validator-api-address string       Listening address (ip and port) for validator-facing traffic proxying the beacon-node API. (default "127.0.0.1:3600")
+
+```
+
+## Host a relay
+
+Relays run a libp2p circuit relay server that allows charon clusters to perform peer discovery and for charon clients behind NAT gateways to be communicated with. If you want to self-host a relay for your cluster(s) the following command will start one.
```
Comment (OisinKyne) [inline docs/charon/charon_cli_reference.md:235]:
```suggestion
Relays run a libp2p [circuit relay](https://docs.libp2p.io/concepts/nat/circuit-relay/) server that allows charon clusters to perform peer discovery and for charon clients behind NAT gateways to be communicated with. If you want to self-host a relay for your cluster(s) the following command will start one.
```

File: docs/charon/charon_cli_reference.md [hunk 8]
```diff
@@ -183,36 +194,81 @@ Usage:
...
+      --log-format string              Log format; console, logfmt or json (default "console")
+      --log-level string               Log level; debug, info, warn or error (default "info")
+      --loki-addresses strings         Enables sending of logfmt structured logs to these Loki log aggregation server addresses. This is in addition to normal stderr logs.
+      --loki-service string            Service label sent with logs to Loki. (default "charon")
+      --monitoring-address string      Listening address (ip and port) for the prometheus and pprof monitoring http server. (default "127.0.0.1:3620")
+      --p2p-allowlist string           Comma-separated list of CIDR subnets for allowing only certain peer connections. Example: 192.168.0.0/16 would permit connections to peers on your local network only. The default is to accept all connections.
+      --p2p-bootnode-relay             Enables using bootnodes as libp2p circuit relays. Useful if some charon nodes are not publicly accessible. Deprecated, should always be enabled. (default true)
+      --p2p-bootnodes strings          Comma-separated list of discv5 bootnode URLs or ENRs. Deprecated, use p2p-relays flag.
```
Comment (OisinKyne) [inline docs/charon/charon_cli_reference.md:256]: `charon relay` is a new command, no one should have the old flags in use, we can clean out these deprecated flags if it's easy to do under the hood.

**Review by OisinKyne — APPROVED** [review]
LGTM. One file to delete, and a question about stripping down the old flags off the `charon relay` command while we have the chance. Although that doesn't have to be in this pr, that is a change to be made over on the charon side.

File: docs/charon/charon_cli_reference.md (hunk 6 above)
Comment (xenowits) [reply docs/charon/charon_cli_reference.md:113]: We have removed makefiles in the cdvc repo as well as in places wherever possible. Haven't encountered any failing tests or builds, checked both cdvc and cdvn repos, charon tests pass, so shouldn't be an issue

File: docs/charon/charon_cli_reference.md (hunk 8 above)
Comment (xenowits) [reply docs/charon/charon_cli_reference.md:256]: We have decided not to remove the deprecated flags in this release but do it in the next release, ie, v0.14.0. This is to ensure we have properly tested the `charon relay` feature

**Pull request merged**